### ISSUE TITLE
React 18 support

### ISSUE THREAD

**Issue by skjnldsv** [scaleflex/ui#16]
`filerobot-image-editor` uses `react 18` and depends on `@scaleflex/ui` and `@scaleflex/icons` which uses `react 16`
This generate a fair amount of warning and/or errors

```js
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @scaleflex/icons@1.0.0-beta.80
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   react@"18.2.0" from filerobot-image-editor@4.3.1
npm WARN   node_modules/filerobot-image-editor
npm WARN     filerobot-image-editor@"^4.3.1" from the root project
npm WARN   9 more (react-clientside-effect, react-dom, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.13.1" from @scaleflex/icons@1.0.0-beta.80
npm WARN node_modules/@scaleflex/icons
npm WARN   @scaleflex/icons@"^1.0.0-beta.80" from @scaleflex/ui@1.0.0-beta.80
npm WARN   node_modules/@scaleflex/ui
npm WARN   1 more (react-filerobot-image-editor)
npm WARN 
npm WARN Conflicting peer dependency: react@16.14.0
npm WARN node_modules/react
npm WARN   peer react@"^16.13.1" from @scaleflex/icons@1.0.0-beta.80
npm WARN   node_modules/@scaleflex/icons
npm WARN     @scaleflex/icons@"^1.0.0-beta.80" from @scaleflex/ui@1.0.0-beta.80
npm WARN     node_modules/@scaleflex/ui
npm WARN     1 more (react-filerobot-image-editor)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @scaleflex/icons@1.0.0-beta.80
npm WARN Found: react-dom@18.2.0
npm WARN node_modules/react-dom
npm WARN   react-dom@"18.2.0" from filerobot-image-editor@4.3.1
npm WARN   node_modules/filerobot-image-editor
npm WARN     filerobot-image-editor@"^4.3.1" from the root project
npm WARN   3 more (react-filerobot-image-editor, react-konva, styled-components)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react-dom@"^16.13.1" from @scaleflex/icons@1.0.0-beta.80
npm WARN node_modules/@scaleflex/icons
npm WARN   @scaleflex/icons@"^1.0.0-beta.80" from @scaleflex/ui@1.0.0-beta.80
npm WARN   node_modules/@scaleflex/ui
npm WARN   1 more (react-filerobot-image-editor)
npm WARN 
npm WARN Conflicting peer dependency: react-dom@16.14.0
npm WARN node_modules/react-dom
npm WARN   peer react-dom@"^16.13.1" from @scaleflex/icons@1.0.0-beta.80
npm WARN   node_modules/@scaleflex/icons
npm WARN     @scaleflex/icons@"^1.0.0-beta.80" from @scaleflex/ui@1.0.0-beta.80
npm WARN     node_modules/@scaleflex/ui
npm WARN     1 more (react-filerobot-image-editor)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @scaleflex/ui@1.0.0-beta.80
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   react@"18.2.0" from filerobot-image-editor@4.3.1
npm WARN   node_modules/filerobot-image-editor
npm WARN     filerobot-image-editor@"^4.3.1" from the root project
npm WARN   9 more (react-clientside-effect, react-dom, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.13.1" from @scaleflex/ui@1.0.0-beta.80
npm WARN node_modules/@scaleflex/ui
npm WARN   @scaleflex/ui@"1.0.0-beta.80" from react-filerobot-image-editor@4.3.1
npm WARN   node_modules/react-filerobot-image-editor
npm WARN 
npm WARN Conflicting peer dependency: react@16.14.0
npm WARN node_modules/react
npm WARN   peer react@"^16.13.1" from @scaleflex/ui@1.0.0-beta.80
npm WARN   node_modules/@scaleflex/ui
npm WARN     @scaleflex/ui@"1.0.0-beta.80" from react-filerobot-image-editor@4.3.1
npm WARN     node_modules/react-filerobot-image-editor
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @scaleflex/ui@1.0.0-beta.80
npm WARN Found: react-dom@18.2.0
npm WARN node_modules/react-dom
npm WARN   react-dom@"18.2.0" from filerobot-image-editor@4.3.1
npm WARN   node_modules/filerobot-image-editor
npm WARN     filerobot-image-editor@"^4.3.1" from the root project
npm WARN   3 more (react-filerobot-image-editor, react-konva, styled-components)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react-dom@"^16.13.1" from @scaleflex/ui@1.0.0-beta.80
npm WARN node_modules/@scaleflex/ui
npm WARN   @scaleflex/ui@"1.0.0-beta.80" from react-filerobot-image-editor@4.3.1
npm WARN   node_modules/react-filerobot-image-editor
npm WARN 
npm WARN Conflicting peer dependency: react-dom@16.14.0
npm WARN node_modules/react-dom
npm WARN   peer react-dom@"^16.13.1" from @scaleflex/ui@1.0.0-beta.80
npm WARN   node_modules/@scaleflex/ui
npm WARN     @scaleflex/ui@"1.0.0-beta.80" from react-filerobot-image-editor@4.3.1
npm WARN     node_modules/react-filerobot-image-editor
```

See https://github.com/scaleflex/filerobot-image-editor/issues/227

**Comment by skjnldsv** [comment]
Awesome! Good luck :)
Feel free to ask if I can be any help! :bow:

**Comment by amrelbialy** [comment]
We are going to fix it soon

**Comment by amrelbialy** [comment]
Fix was added in v2.5.0, it will be updated soon in filerobot-image-editor
